lib/connectors/gitlab/connector.rb

```diff
@@ -5,66 +5,101 @@
 #
 
 # frozen_string_literal: true
+require 'active_support/core_ext/hash/indifferent_access'
 
 require 'connectors/base/connector'
+require 'connectors/gitlab/extractor'
 require 'connectors/gitlab/custom_client'
 require 'connectors/gitlab/adapter'
-require 'connectors/gitlab/config'
-require 'connectors/gitlab/extractor'
-require 'rack/utils'
+require 'utility/sink'
+require 'app/config'
 
 module Connectors
   module GitLab
     class Connector < Connectors::Base::Connector
       SERVICE_TYPE = 'gitlab'
 
+      def initialize
+        super()
+        @extractor = Connectors::GitLab::Extractor.new(
+          :base_url => configurable_fields[:base_url][:value],
+          :api_token => configurable_fields[:api_token][:value]
+        )
+        @sink = Utility::Sink::CombinedSink.new(
+          [Utility::Sink::ConsoleSink.new,
+           Utility::Sink::ElasticSink.new(App::Config[:connector_package_id])]
+        )
+      end
+
       def display_name
         'GitLab Connector'
       end
 
       def configurable_fields
-        {
-          'api_token' => {
-            'label' => 'API Token',
-            'value' => nil
+        @configurable_fields ||= {
+          :api_token => {
+            :label => 'API Token',
+            :value => App::Config[:gitlab][:api_token]
```

Comment:
`API token` and `Base URL` should come from connector configuration (i.e. `.elastic-connectors`), instead of YAML configuration.